src/protocols/bdx/AsyncTransferFacilitator.h

```diff
@@ -0,0 +1,130 @@
+     */
+    virtual void HandleTransferSessionOutput(TransferSession::OutputEvent & event) = 0;
+
+    /**
+     * This method should be implemented to destroy the object subclassing AsyncTransferFacilitator.
+     */
+    virtual void DestroySelf() = 0;
+
+protected:
+    CHIP_ERROR OnMessageReceived(Messaging::ExchangeContext * ec, const PayloadHeader & payloadHeader,
+                                 System::PacketBufferHandle && payload) override;
+    void OnResponseTimeout(Messaging::ExchangeContext * ec) override;
+    void OnExchangeClosing(Messaging::ExchangeContext * ec) override;
+
+    CHIP_ERROR SendMessage(const TransferSession::MessageTypeData msgTypeData, System::PacketBufferHandle & msgBuf);
+
+    static bdx::StatusCode GetBdxStatusCodeFromChipError(CHIP_ERROR err);
+
+    void CleanUp();
+
+    void HandleNextOutputEvents();
+
+    // The transfer session coresponding to this AsyncTransferFacilitator object.
+    TransferSession mTransfer;
+
+    // The Exchange holder that holds the exchange context used for sending and receiving BDX messages.
+    Messaging::ExchangeHolder mExchange;
+
+    // The timeout for the BDX transfer session.
+    System::Clock::Timeout mTimeout;
+
+    System::Layer * mSystemLayer;
+
+private:
+    bool mHandlingOutputEvents;
+};
+
+/**
+ * An AsyncTransferFacilitator that is initialized to respond to an incoming BDX transfer request.
+ *
+ * Provides a method for initializing the TransferSession members but still needs to be extended to implement
+ * HandleTransferSessionOutput.
+ *
+ * An instance of some subclass of this class should be used as the exchange delegate for a BDX transfer.
+ */
+class AsyncResponder : public AsyncTransferFacilitator
+{
+public:
+    /**
+     * Initialize the TransferSession state machine to be ready for an incoming transfer request.
+     *
+     * @param[in] exchangeCtx     The exchange to use for the transfer.
+     * @param[in] role            The role of the Responder: Sender or Receiver of BDX data
+     * @param[in] xferControlOpts Supported transfer modes (see TransferControlFlags)
+     * @param[in] maxBlockSize    The maximum supported size of BDX Block data
+     * @param[in] timeout         The chosen timeout delay for the BDX transfer
+     */
+    CHIP_ERROR PrepareForTransfer(System::Layer * layer, Messaging::ExchangeContext * exchangeCtx, TransferRole role,
+                                  BitFlags<TransferControlFlags> xferControlOpts, uint16_t maxBlockSize,
+                                  System::Clock::Timeout timeout);
+
+    /**
+     * This is called by the subclass implementing HandleTransferSessionOutput to notify the AsyncTransferFacilitator
+     * that it has handled the OutputEvent specified in event and returns an error code (if any) or success in the error paramter.
+     * Once this is called the AsyncTransferFacilitator either aborts the transfer if an error has ocurred or drives the
```

Comment:
Again, this is the AsyncResponder, not the AsyncTransferFacilitator.

Does this need to be documented in the API?  Does the API consumer (subclass) need to know what happens when it calls this function?

Reply:
no. i will remove additional details

Reply:
will fix the AsyncTransferFacilitator-> AsyncResponder

Reply:
done.